### DISCUSSION TITLE
Capturing the endpoint result from another component in Rtk query

### DISCUSSION THREAD

**Discussion by phoique** [reduxjs/redux-toolkit#2310] [Q&A] [answered]
Hello,
 
I have a rtk query de queries service. I will make the request in a different component and the display of the result in a different component. Here is how I can capture the result of the endpoint in the other component. Passing it as props, sending requests in the same way as the body (receiving data from the cache) did not come a good solution. Is there such a thing as result sharing between hooks?

**Comment by phryneas** [comment, marked as answer]
Use the same hook with the same argument in another component and no extra request will be made.

**Reply by phoique** [reply]
Yes I know that. Do you think this is a good solution? My arguments are mixed, so their body varies. For example, while the locations value is an array on page a, I send lat lon value on page b. I get the lat lon value by reading it from the state.

**Reply by phryneas** [reply]
I might just not be getting your question then. Could you please explain what you are asking for?

**Reply by phoique** [reply]
Well, I will try to explain it in detail for you. I have 1 endpoint and this endpoint is used on 3 pages. I also have a component called statistics. This component is used in common on 3 pages. It converts the results returned from the endpoint to statistics and displays them. What I want is to get the last positive result of the request sent to the relevant endpoint in that page in the statistics component. I can't pass it off as component props because it's too far away. Is there any solution other than sending request again with the same arguments?

**Reply by phryneas** [reply]
You will have to transport the argument using something like Redux store then. There is no support for a "last query" because in so many use cases that would instantly break - if two components render the same query with a different argument at the same time, which one is the last query? The one that rerendered last? It would be completely random.

**Reply by phoique** [reply]
I understand, thank you for your interest, I will move the production to redux as you said and analyze it.
